packages/react/src/components/organisms/GenTeaser/index.js

```diff
@@ -15,708 +15,57 @@
-import TeaserSearch from 'MayflowerReactGenTeaser/TeaserSearch';
-import TeaserOrgs from 'MayflowerReactGenTeaser/TeaserOrgs';
-import { buildUrl } from 'MayflowerReactOrganisms/GenTeaser/utils';
-
-const GenTeaser = (props) => {
-  const {
-    children, onClick, onKeyDown, stacked, align, className, ...rest
-  } = props;
-  const teaserClasses = classNames({
-    'ma__gen-teaser': true,
-    'ma__gen-teaser--clickable': onClick,
-    'ma__gen-teaser--stacked': stacked,
-    [`ma__gen-teaser--align-${align}`]: align,
-    [`${className}`]: !!className
-  });
-  const role = onClick ? 'button' : null;
-  return(
-    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
-    <section className={teaserClasses} onClick={onClick} onKeyDown={onKeyDown} role={role} {...rest}>
-      {children}
-    </section>
-  );
-};
-
-GenTeaser.displayName = 'GenTeaser';
-
-GenTeaser.propTypes = {
-  /** A custom on click function */
-  onClick: PropTypes.func,
-  /** A custom on key down function */
-  onKeyDown: PropTypes.func,
-  /** React children to render */
-  children: PropTypes.node,
-  /** whether to stack image on top */
-  stacked: PropTypes.bool,
-  /** alignment for description relative to image */
-  align: PropTypes.oneOf(['top', 'center']),
-  /** A custom class. */
-  className: PropTypes.string
-};
+import Container from 'MayflowerReactGenTeaser/GenTeaserContainer';
+import Details from 'MayflowerReactGenTeaser/GenTeaserDetails';
+import Image from 'MayflowerReactGenTeaser/GenTeaserImage';
+import Eyebrow from 'MayflowerReactGenTeaser/GenTeaserEyebrow';
+import Button from 'MayflowerReactGenTeaser/GenTeaserButton';
+import Stat from 'MayflowerReactGenTeaser/GenTeaserStat';
+import Title from 'MayflowerReactGenTeaser/GenTeaserTitle';
+import Emphasis from 'MayflowerReactGenTeaser/GenTeaserEmphasis';
+import Date from 'MayflowerReactGenTeaser/GenTeaserDate';
+import Orgs from 'MayflowerReactGenTeaser/GenTeaserOrgs';
+import Description from 'MayflowerReactGenTeaser/GenTeaserDescription';
+import SearchBar from 'MayflowerReactGenTeaser/GenTeaserSearchBar';
+import SubLinks from 'MayflowerReactGenTeaser/GenTeaserSubLinks';
+import KeyAction from 'MayflowerReactGenTeaser/GenTeaserKeyAction';
+import MoreInfo from 'MayflowerReactGenTeaser/GenTeaserMoreInfo';
+import PrimaryInfo from 'MayflowerReactGenTeaser/GenTeaserPrimaryInfo';
+import SecondaryInfo from 'MayflowerReactGenTeaser/GenTeaserSecondaryInfo';
+import Address from 'MayflowerReactGenTeaser/GenTeaserAddress';
+import Phone from 'MayflowerReactGenTeaser/GenTeaserPhone';
+import Email from 'MayflowerReactGenTeaser/GenTeaserEmail';
+import Event from 'MayflowerReactGenTeaser/GenTeaserEvent';
+import InfoDetails from 'MayflowerReactGenTeaser/GenTeaserInfoDetails';
+import Tags from 'MayflowerReactGenTeaser/GenTeaserTags';
+
+const GenTeaser = (props) => (
+  <Container {...props} />
+);
+
+GenTeaser.propTypes = Container.propTypes;
+
+GenTeaser.Details = Details;
```

Comment:
Does this file mean that someone can still implement GenTeaser with subcomponents if they opt-in to import from the index?

Reply:
If so, can you please add to the documentation outlining the two options? And maybe add some explanation of why we did what we did here and that this is the recommended import option because it excludes unused subcomponents.

Reply:
Just added the documentation about the previous approach.